modules/xmpp/xmpp.js

```diff
@@ -509,4 +512,45 @@ export default class XMPP extends Listenable {
         initRayo();
         initStropheLogger();
     }
+
+    /**
+     * Returns details about connection failure. Shard change or is it after
+     * suspend.
+     * @returns {object} contains details about a connection failure.
+     * @private
+     */
+    _getConnectionFailedReasonDetails() {
+        const details = {};
+
+        // check for moving between shard if information is available
+        if (this.options.deploymentInfo
+            && this.options.deploymentInfo.shard
+            && this.connection._proto
+            && this.connection._proto.lastResponseHeaders) {
+
+            // split headers by line
+            const headersArr = this.connection._proto.lastResponseHeaders
+                .trim().split(/[\r\n]+/);
+            const headers = {};
+
+            headersArr.forEach(line => {
+                const parts = line.split(': ');
+                const header = parts.shift();
+                const value = parts.join(': ');
+
+                headers[header] = value;
+            });
+
+            /* eslint-disable camelcase */
+            details.shard_changed
+                = this.options.deploymentInfo.shard
+                    !== headers['x-jitsi-shard'];
```

Comment:
I see "X-Jitsi-Shard" coming from meet.jit.si, and [HTTP headers seem to be case-insensitive](https://tools.ietf.org/html/rfc7230#section-3.2). Is there something in strophe that converts them to all lowercase?

Reply:
Hum, I suppose the XHR API works that way. All strophe does is: req.xhr.getAllResponseHeaders();
https://github.com/strophe/strophejs/commit/93d63d3f7adc1ea126e6ed57f1b5fabf6a34c765#diff-c8241e10984282f97b0a3814d0bb41e7R619
And I've been testing this with meet.jit.si.